yt_dlp/extractor/ximalaya.py

```diff
@@ -82,6 +114,24 @@ def _real_extract(self, url):
             audio_info_file, audio_id,
             f'Downloading info json {audio_info_file}', 'Unable to download info file')
 
+        # NOTE(xcsong): VIP-restricted audio
+        if audio_info.get('is_paid'):
+            ts = int(time.time())
+            audio_info_file = f'{scheme}://mpay.ximalaya.com/mobile/track/pay/{audio_id}/{ts}?device=pc&isBackend=true&_={ts}'
+            audio_info = self._download_json(
+                audio_info_file, audio_id,
+                f'Downloading VIP info json {audio_info_file}', 'Unable to download VIP info file')
+            filename = self._decrypt_filename(audio_info)
+            sign, token, timestamp = self._decrypt_url_params(audio_info)
+            buy_key = audio_info.get('buyKey')
+            duration = audio_info.get('duration')
+            domain = audio_info.get('domain')
+            api_version = audio_info.get('apiVersion')
+            args = f'?sign={sign}&buy_key={buy_key}&token={token}&timestamp={timestamp}&duration={duration}'
+            audio_info['play_path_64'] = f'{domain}/download/{api_version}{filename}{args}'
```

Comment:
I would write this like so:
```suggestion
            audio_info['play_path_64'] = update_url_query(
                f'{audio_info['domain']}/download/{audio_info['apiVersion']}{filename}', {
                    'sign': sign,
                    'token': token,
                    'timestamp': timestamp,
                    'buy_key': audio_info['buyKey'],
                    'duration': audio_info['duration'],
                })
```
But do we really want to overwrite `play_path_64` or should this just be a separately extracted format?
I am not sure the audio tracks are actually 64 kbps. Maybe just not set the bitrate and instead set the filesize based on `totalLength`

Reply:
Yes, the audio might not be either 64 kbps or 24 kbps. I agree that we can establish a new bitrate based on the formula: `totalLength * 8 / duration`, which equals approximately 48.8 kbps in the example provided.

```json
{
  "ret": 0,
  "msg": "0",
  "trackId": 749837049,
  "uid": 37504890,
  "albumId": 83105493,
  "title": "第1集 雷旭",
  "domain": "http://audiopay.cos.tx.xmcdn.com",
  "totalLength": 4833714,
  "sampleDuration": 180,
  "sampleLength": 1156925,
  "isAuthorized": true,
  "apiVersion": "1.0.0",
  "seed": 9967,
  "fileId": "40*42*2*17*28*1*32*40*33*67*7*21*28*35*28*54*25*31*2*50*28*38*33*18*9*33*18*6*33*45*13*51*52*14*64*14*13*24*61*3*38*6*18*8*65*40*1*41*47*29*65*22*7*12*24*21*28*",
  "buyKey": "FM",
  "duration": 792,
  "ep": "qz91a9wrGyYFgfkIb9tN5FsKBElygSTGTdUVfA3mC377mb6Q3/+Bm/AP1f2J0KdhXeJ2",
  "highestQualityLevel": 2,
  "downloadQualityLevel": 1,
  "authorizedType": 1,
  "volumeGain": -5.4
}
```